draft-ietf-privacypass-rate-limit-tokens.md

```diff
@@ -1501,13 +1517,49 @@ at specific Clients or sets of Clients.
 As with the basic issuance protocol {{ISSUANCE}}, the token_key_id is truncated to a single
 octet to mitigate the risk of unique keys per client.
 
-To mitigate the risk of a targeted Issuer Encapsulation Key, the Attester can observe and validate
-the issuer_encap_key_id presented by the Client to the Issuer. As described in {{request-one}}, Attesters
-MUST validate that the issuer_encap_key_id in the Client's TokenRequest matches a known Issuer
+Clients SHOULD prevent targeting attacks by checking the consistency of the encapsulation
+key across time and across other clients. Consistency for privacy pass keys is described in
+{{?CONSISTENCY=I-D.ietf-privacypass-key-consistency}}.
+{{?CONSISTENCY-MIRROR=I-D.group-privacypass-consistency-mirror}} describes a specific
+approach in which Clients can check the consistency of a key using a mirror server.
+Since encapsulation keys are available in the Issuer well-known configuration ({{setup}}),
+the Client can fetch the configuration via a mirror to perform a consistency check.
+
+Clients can also detect inconsistency if the encapsulation key changes across multiple
+challenges (indicating that an Origin might be trying to target Clients, but did not
+recognize the Client across two requests). If such changes are detected multiple times
+within a short period of time, Clients MUST NOT fetch more tokens using the key.
+
+The Attester can also help ensure consistency with an in-band check, which conforms
+to the approach in {{?CONSISTENCY-INBAND=I-D.pw-privacypass-in-band-consistency}}.
+The Attester can observe and validate the issuer_encap_key_id presented by the Client to
+the Issuer. As described in {{request-one}}, Attesters MUST validate that the
+issuer_encap_key_id in the Client's TokenRequest matches a known Issuer
 Encapsulation Key public key for the Issuer. The Attester needs to support key rotation, but
 ought to disallow very rapid key changes, which could indicate that an Origin is colluding with
 an Issuer to try to rotate the key for each new Client in order to link the client activity.
 
+## Client Identification with Unique Per-Origin Token Keys
+
+Client activity could also be linked if an Origin and Issuer collude to use a unique
+key for the Token Keys, which are per-origin.
+
+Since Attesters do not see per-origin identities that can be correlated across Clients,
+Attesters cannot perform in-band consistency checks.
+
+Clients SHOULD prevent targeting attacks by checking the consistency of token keys
+key across time and across other clients. Consistency for privacy pass keys is described in
+{{?CONSISTENCY=I-D.ietf-privacypass-key-consistency}}.
+{{?CONSISTENCY-MIRROR=I-D.group-privacypass-consistency-mirror}} describes a specific
+approach in which Clients can check the consistency of a key using a mirror server.
+Since per-origin token keys are available in the Issuer well-known configuration ({{setup}}),
```

Comment:
```suggestion
Clients SHOULD check encapsulation key consistency to help mitigate targeting attacks.
Consistency for Privacy Pass resources such as Issuer directories and token keys is described in
{{?CONSISTENCY=I-D.ietf-privacypass-key-consistency}}.
{{?CONSISTENCY-MIRROR=I-D.group-privacypass-consistency-mirror}} describes a specific
approach in which Clients can check the consistency of a key using a mirror server.
Since per-origin token keys are available in the Issuer well-known configuration ({{setup}}),
```